model/src/pyrenew/latent/hospitaladmissions.py

```diff
@@ -184,25 +184,33 @@ def sample(
 
         latent_hospital_admissions = jnp.convolve(
             infection_hosp_rate_t,
-            infection_to_admission_interval,
+            infection_to_admission_interval.value,
             mode="full",
         )[: infection_hosp_rate_t.shape[0]]
 
         # Applying the day of the week effect
         latent_hospital_admissions = (
             latent_hospital_admissions
-            * self.day_of_week_effect_rv(**kwargs)[0]
+            * self.day_of_week_effect_rv(
+                n_timepoints=latent_hospital_admissions.size, **kwargs
+            )[0].value
         )
 
         # Applying probability of hospitalization effect
         latent_hospital_admissions = (
-            latent_hospital_admissions * self.hosp_report_prob_rv(**kwargs)[0]
+            latent_hospital_admissions
+            * self.hosp_report_prob_rv(**kwargs)[0].value
         )
 
         npro.deterministic(
             self.latent_hospital_admissions_varname, latent_hospital_admissions
         )
 
         return HospitalAdmissionsSample(
-            infection_hosp_rate, latent_hospital_admissions
+            infection_hosp_rate=infection_hosp_rate,
+            latent_hospital_admissions=SampledValue(
+                value=latent_hospital_admissions,
+                t_start=self.infection_to_admission_interval_rv.t_start,
+                t_unit=self.infection_to_admission_interval_rv.t_unit,
```

Comment:
I don't think these are the relevant t_start/t_unit.

I would have thought they should be the same as the RV itself (or, eventually computed dynamically from the values in a time-aware `latent_infections` that gets passed + attributes of the delay interval, if we eventually go in that direction). For now, suggest:

```suggestion
                t_start=self.t_start,
                t_unit=self.t_unit,
```